organizer/qorganizer-eds-saverequestdata.cpp

```diff
@@ -33,6 +33,14 @@ SaveRequestData::SaveRequestData(QOrganizerEDSEngine *engine,
     // map items by source
     Q_FOREACH(const QOrganizerItem &i, request<QOrganizerItemSaveRequest>()->items()) {
         QByteArray sourceId = i.collectionId().localId();
+        if (sourceId.isNull()) {
+            /* We use an empty bytearray to indicate that the item has no
+             * associated collection; we cannot leave it to null here, as the
+             * caller of nextSourceId() would consider that as having reached
+             * the end of the list.
+             */
```

Comment:
I'm not whether i agree on that comment...
After the call of `nextSourceId` at qorganizer-eds-engine.cpp:567 it is checked `sourceId` is null, but also if `data->end()` is true too `data->finish()` gets called.
But on the other hand `data->finish()` gets called at qorganizer-eds-engine.cpp:163 if `sourceId.isEmpty()`. Is that really consistent and correct?
Should the comment refer to qorganizer-eds-engine.cpp:585, where the default collection is assumed when the `sourceId` is empty? Could we theoretically change it to `isNull` there instead of setting it to `""` here?